scripts/components/forms/index.es6

```diff
@@ -0,0 +1,158 @@
+import Filters from 'components/forms/filters';
+import FormsListItem from 'components/forms/list-item';
+import List from 'react-list';
+import React from 'react';
+
+var PER_PAGE = 10;
+
+export default React.createClass({
+  mixins: [Cursors],
+
+  propTypes: {
+    portalId: React.PropTypes.number
+  },
+
+  getDefaultProps: function () {
+    return {
+      category: '',
+      forms: [],
+      filtersAreShowing: true,
+      query: '',
+      searchableAttributes: ['name']
+    };
+  },
+
+  getInitialState: function () {
+    return {
+      forms: this.props.forms,
+      category: this.props.category,
+      query: this.props.query
+    };
+  },
+
+  componentWillMount: function () {
+    if (this.state.forms.length) this.sortAndUpdate(this.state.forms);
+  },
+
+  fetch: function (cb) {
+    api.get('/portals/:portal_id/forms', {
+      portal_id: this.props.portalId,
+      page: Math.floor(this.state.forms.length / PER_PAGE) + 1,
+      per_page: PER_PAGE
+    }, _.partial(this.handleFetch, cb));
+  },
+
+  handleFetch: function (cb, er, res) {
+    if (er) return cb(er);
+    this.update({
+      forms: {$set: _.unique(this.state.forms.concat(res.data), 'id')}
+    });
+    cb(null, res.data.length < PER_PAGE);
+  },
+
+  matchesCategory: function(form) {
+    var a = this.state.category;
+    var b = form.category.name;
+    return !a || a === b;
+  },
+
+  matchesQuery: function(form) {
+    var query = this.state.query;
+    if (!query) return true;
+    var words = _str.words(query.toLowerCase());
+    var searchableWords = this.searchableWordsFor(form);
+    return _.every(words, function (wordA) {
+      return _.any(searchableWords, function (wordB) {
+        return _str.startsWith(wordB, wordA);
+      });
+    });
+  },
+
+  searchableWordsFor: function (form) {
+    return _str.words(
+      _.values(_.pick(form, this.props.searchableAttributes))
+      .join(' ')
+      .toLowerCase()
+    );
+  },
+
+  formMatchesFilters: function(form) {
+    return (
+      this.matchesCategory(form) &&
+      this.matchesQuery(form)
+    );
+  },
+
+  getFilteredForms: function() {
+    return this.state.forms.filter(this.formMatchesFilters);
+  },
+
+  renderFilters: function (forms) {
+    if (!this.state.forms.length || !this.props.filtersAreShowing) return;
+    return (
+      <Filters
+        forms={forms}
+        cursors={{
+          query: this.getCursor('query'),
+          category: this.getCursor('category')
+        }}
+      />
+    );
+  },
+
+  renderListItem: function (form) {
+    var i = this.state.forms.indexOf(form);
+    return (
+      <FormsListItem
+        key={form.id}
+        cursors={{form: this.getCursor('forms', i)}}
+      />
+    );
+  },
+
+  renderLoading: function () {
+    return <div className='osw-inset-block'>Loading...</div>;
+  },
+
+  renderError: function (er) {
+    return (
+      <div className='osw-inset-block osw-inset-block-red'>{er.toString()}</div>
```

Comment:
Kinda seems like this DOM + classes for the error div and for the loading div should be abstracted out, but that's another project for another day.